modmesh/apputil.py

```diff
@@ -70,21 +70,32 @@ def __init__(self, name):
         # Each run of the application appends a new environment.
         environ[name] = self
 
-        # default value will be overwritten when `run_code` is called
-        self._redirectStdOutFile = "stdout.txt"
-        self._redirectStdErrFile = "stderr.txt"
+        self._stdoutFd = -1
+        self._stderrFd = -1
 
     def run_code(self, code):
-        with open(self._redirectStdOutFile , 'w') as f1:
-            with contextlib.redirect_stdout(f1):
-                with open(self._redirectStdErrFile , 'w') as f2:
-                    with contextlib.redirect_stderr(f2):
-                        try:
-                            exec(code, self.globals, self.locals)
-                        except Exception as e:
-                            print(("{}: {}".format(type(e).__name__, str(e))), file=sys.stderr)
-                            print("traceback:", file=sys.stderr)
-                            traceback.print_stack()
+        import msvcrt
+
+        oldStdoutFd = sys.stdout
+        oldStderrFd = sys.stderr
+        try:
+            if self._stdoutFd <= 0:
+                raise Exception("wrong stdout fd:", self._stdoutFd)
+            if self._stderrFd <= 0:
+                raise Exception("wrong stderr fd:", self._stderrFd)
+
+            sys.stdout = msvcrt.get_osfhandle(self._stdoutFd)
+            sys.stderr = msvcrt.get_osfhandle(self._stderrFd)
+            exec(code, self.globals, self.locals)
+            
+            sys.stdout = oldStdoutFd
+            sys.stderr = oldStderrFd
+        except Exception as e:
+            sys.stdout = oldStdoutFd
+            sys.stderr = oldStderrFd
+            sys.stderr.write(("{}: {}".format(type(e).__name__, str(e))))
```

Comment:
```suggestion
            sys.stderr.write(f"{type(e).__name__}: {e}")
```
There might be something in the `traceback` module that does this for you already.